src/logging.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <google/protobuf/stubs/common.h>
+
+#if GOOGLE_PROTOBUF_VERSION < 3021012
+#define USE_ABSL_LOGGING 0
+#else
+#define USE_ABSL_LOGGING 1
+#endif
+
+#include <grpc/support/log.h>
+
+#if USE_ABSL_LOGGING
+#include <absl/log/globals.h>
+#include <absl/log/initialize.h>
+#include <absl/log/log_sink_registry.h>
+#else
+#include <google/protobuf/stubs/logging.h>
+#endif
+
+#include "ngx_str.hpp"
+
+#if USE_ABSL_LOGGING
+
+class AbslLogSink : public absl::LogSink {
+public:
+    AbslLogSink() { absl::AddLogSink(this); }
+    ~AbslLogSink() override { absl::RemoveLogSink(this); }
+
+    void Send(const absl::LogEntry& entry) override
+    {
+        auto severity = entry.log_severity();
+
+        ngx_uint_t level =
+            severity == absl::LogSeverity::kFatal   ? NGX_LOG_EMERG :
+            severity == absl::LogSeverity::kError   ? NGX_LOG_ERR :
+            severity == absl::LogSeverity::kWarning ? NGX_LOG_WARN :
+                      /*absl::LogSeverity::kInfo*/    NGX_LOG_INFO;
+
+        auto message = toNgxStr(entry.text_message());
+
+        ngx_log_error(level, ngx_cycle->log, 0, "OTel/protobuf: %V", &message);
+    }
+};
+
+std::unique_ptr<AbslLogSink> gAbslLogSink;
```

Comment:
Such globals in header might cause link errors.

Reply:
See [ccb30e1](https://github.com/nginxinc/nginx-otel/pull/34/commits/ccb30e11659828c2fb9efa0229cecee0cdcb4237).